ccmlib/cluster.py

```diff
@@ -702,3 +702,31 @@ def timed_grep_nodes_for_patterns(self, versions_to_patterns, timeout_seconds, f
 
     def wait_for_any_log(self, pattern, timeout, filename='system.log', marks=None):
         return common.wait_for_any_log(self.nodelist(), pattern, timeout, filename=filename, marks=marks)
+
+    def show_logs(self, selected_nodes_names=()):
```

Comment:
This is a surprising python gotcha, but one should never use `()` or []` as the default value for a keyword argument. Everytime this method is called, you don't get a new empty tuple, but end up re-using the same tuple from previous calls. It is recommended to instead:
```
def show_logs(self, selected_node_names=None):

    if selected_node_names is None:
        selected_node_names = ()
....
```

Reply:
OK, I fix this. Thank you.